File: Assignment/GroupAssignment/backend/src/main/java/com/example/backend/dto/GuestBranchMenuItemDTO.java
package com.example.backend.dto;

import java.math.BigDecimal;
import java.util.UUID;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class GuestBranchMenuItemDTO {

    private UUID branchMenuItemId;
    private UUID branchId;
    private UUID menuItemId;
    private boolean available;
    private String name;
    private String description;
    private BigDecimal price;
    private BigDecimal discountedPrice;
    private UUID categoryId;
    private String imageUrl;
    private boolean bestSeller;
    private boolean hasCustomization;

}

File: Assignment/GroupAssignment/frontend/src/types/dto/guest-branch-menu-item.dto.ts
export interface GuestBranchMenuItemDTO {
  branchMenuItemId: string;
  branchId: string;
  menuItemId: string;
  available: boolean;
  name: string;
  description: string;
  price: number;
  discountedPrice?: number;
  categoryId: string;
  imageUrl?: string;
  bestSeller: boolean;
  hasCustomization?: boolean;
}
